feat: Basic metrics for minidump scrubbing

flub opened the pull request:
A dump counter keeping track of minidump scrubbing.

I'm considering this as part of the larger minidump scrubbing work, so:

#skip-changelog

Commit:
feat: Basic metrics for minidump scrubbing

jan-auer commented on relay-server/src/actors/events.rs:
```diff
@@ -1019,8 +1019,17 @@ impl EventProcessor {
+                    metric!(
+                        counter(RelayCounters::MinidumpsScrubbed) += 1,
+                        status = "ok"
+                    );
```
This will also increment even if no rules were applied. Should we track this?

flub replied:
Good point, I've improved this now.

Commit:
Only cound scrubbed minidumps

The downside is that the two status variants no longer simply add up.

Commit:
Make numbers add up again

Introducing a new status makes this add up again.

Commit:
Change this to a timer

Timers can also be counted.  And this allows us to check the impact of
the scrubbing better.

untitaker commented on relay-server/src/actors/events.rs:
```diff
@@ -1018,9 +1018,22 @@ impl EventProcessor {
+                            status = "error"
+                        );
+                        log::warn!("failed to scrub minidump: {}", LogError(&scrub_error));
+                        processor.scrub_attachment(filename, &mut payload);
```
Can we also time this? Should be easier since this is infallible and the metric macro has block syntax

flub replied:
I kind of wanted to time this inside `scrub_attachments` but that's in the general crate and doesn't have the metrics.  Now we'll have to remember to add this metric every time we call it :(

Commit:
Time attachments as well
